Atlassian.Jira/Remote/JiraRestClient.cs

```diff
@@ -195,7 +198,8 @@ private JToken GetValidJsonFromResponse(IRestRequest request, IRestResponse resp
 
             if (!string.IsNullOrEmpty(response.ErrorMessage))
             {
-                throw new InvalidOperationException($"Error Message: {response.ErrorMessage}\nContent: {content}\nCode: {response.StatusCode}", response.ErrorException);
+                var headers = PrintResponseAsync(request).Result;

```

Comment:
Never use .Result without await first

Reply:
Thought `.Result` will block... My mistake...
Will this work?
```csharp
if (!string.IsNullOrEmpty(response.ErrorMessage))
{
    var task = PrintResponseAsync(request);
    task.Wait();
    var headers = task.Result;
    throw new InvalidOperationException($"Error Message: {response.ErrorMessage}\nContent: {content}\nCode: {response.StatusCode}\nHeaders: {headers}", response.ErrorException);
}
```

Never called an async method from a synced method

Reply:
Why not 
var msg = await task;

Reply:
this method is not async, and I think turning it to async might have a ripple effect, but I can try if you think it's important enough

Reply:
Spoke with @YonatanEldar and we think that this:
```csharp
string headers = null;
try
{
    headers = PrintResponseAsync(request).Result;
}
catch
{
    // ignored
}

throw new InvalidOperationException(
    $"Error Message: {response.ErrorMessage}\nContent: {content}\nCode: {response.StatusCode}\nHeaders: {headers}",
    response.ErrorException);
```
Should be good enough...
WDYT?

Reply:
trusting you that the ripple effect is too big to be worth it

Reply:
😄